src/main/scala/analysis/InterprocSteensgaardAnalysis.scala

```diff
@@ -0,0 +1,395 @@
+                    val alpha = FreshVariable()
+                    X2_star.foreach(
+                      x => unify(exprToStTerm(x), PointerRef(alpha))
+                    )
+                    unify(alpha, varToStTerm(RegisterVariableWrapper(X1)))
+
+                    println("Memory load: " + memoryLoad)
+                    println("Index: " + memoryLoad.index)
+                    println("X2_star: " + X2_star)
+                    println("X1: " + X1)
+                    println("LocalAssign: " + localAssign)
+
+                    // TODO: This might not be correct for globals
+                    // X1 = &X: [[X1]] = ↑[[X2]] (but for globals)
+                    val $X2 = exprToRegion(memoryLoad.index, cmd)
+                    $X2.foreach(
+                      x => unify(varToStTerm(RegisterVariableWrapper(localAssign.lhs)), PointerRef(allocToTerm(x)))
+                    )
+                  case variable: Variable =>
+                    // X1 = X2: [[X1]] = [[X2]]
+                    val X1 = localAssign.lhs
+                    val X2 = variable
+                    unify(varToStTerm(RegisterVariableWrapper(X1)), varToStTerm(RegisterVariableWrapper(X2)))
+                }
+            }
+          case memoryAssign: MemoryAssign =>
+            // *X1 = X2: [[X1]] = ↑a ^ [[X2]] = a where a is a fresh term variable
+            val X1_star = exprToRegion(memoryAssign.rhs.index, cmd)
+            val X2 = evaluateExpressionWithSSA(memoryAssign.rhs.value, constantProp(n))
+            var possibleRegions = Set[MemoryRegion]()
+            if X2.isEmpty then
+              println("Maybe a region: " + exprToRegion(memoryAssign.rhs.value, cmd))
+              possibleRegions = exprToRegion(memoryAssign.rhs.value, cmd)
+            println("X2 is: " + X2)
+            println("Evaluated: " + memoryAssign.rhs.value)
+            println("Region " + X1_star)
+            println("Index " + memoryAssign.rhs.index)
+            val alpha = FreshVariable()
+            X1_star.foreach(
+              x =>
+                unify(exprToStTerm(x), PointerRef(alpha))
+                x.content.addAll(X2)
+                x.content.addAll(possibleRegions.filter(r => r != x))
+            )
+            X2.foreach(
+              x => unify(alpha, exprToStTerm(x))
+            )
+            possibleRegions.foreach(
+              x => unify(alpha, exprToStTerm(x))
+            )
+          case _ => // do nothing TODO: Maybe LocalVar too?
+        }
+      case _ => // do nothing
+    }
+  }
+
+  private def unify(t1: Term[StTerm], t2: Term[StTerm]): Unit = {
+    //Logger.info(s"univfying constraint $t1 = $t2\n")
+    solver.unify(
+      t1,
+      t2
+    ) // note that unification cannot fail, because there is only one kind of term constructor and no constants
+  }
+
+  type PointsToGraph = Map[Object, Set[VariableWrapper | MemoryRegion]]
+
+  /** @inheritdoc
+    */
+  def pointsTo(): PointsToGraph = {
```

Comment:
Why does this need its own type definition when this is the only use? Why does it not just correctly have CfgNode instead of Object?